### ISSUE TITLE
Unicode-related panic when editing comment

### ISSUE THREAD

**Issue by jturner314** [rust-analyzer/rust-analyzer#7635]
If I create a new project with the following `Cargo.toml` file:

```toml
[package]
name = "ra_test"
version = "0.1.0"
edition = "2018"

[dependencies]
```

and `main.rs` file:

```rust
/// 𝔼
fn main() {}
```

then, if I place the cursor after the 𝔼 in `main.rs` and type any letter, rust-analyzer panics. The standard error recorded by Emacs in `*rust-analyzer::stderr*` is the following:

```
Panic context:
> notification: textDocument/didChange

thread 'main' panicked at 'assertion failed: self.is_char_boundary(n)', /build/rust/src/rustc-1.49.0-src/library/alloc/src/string.rs:1556:29
stack backtrace:
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: "SendError(..)"', /build/.cargo/registry/src/github.com-1ecc6299db9ec823/lsp-server-0.5.0/src/stdio.rs:29:37
stack backtrace:
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

Process rust-analyzer stderr finished
```

I'm using rust-analyzer with Emacs and a somewhat outdated `lsp-mode` package. I installed rust-analyzer using the Arch Linux package:

```
$ rust-analyzer --version
rust-analyzer 336909b

$ pacman -Qi rust-analyzer
Name            : rust-analyzer
Version         : 20210208-1
...
```

Please let me know if you have any trouble reproducing the issue or if any additional information would be helpful. (For example, I could probably figure out how to run rust-analyzer with `RUST_BACKTRACE=full`.)

**Comment by lnicola** [comment]
That's probably an emacs issue: https://github.com/rust-analyzer/rust-analyzer/issues/4263#issuecomment-762781674

**Comment by jturner314** [comment]
Ah, yes, this does look like a duplicate, and it looks like the lsp-mode issue is being addressed in emacs-lsp/lsp-mode#2080. I'm sorry for the noise here.
